### ISSUE TITLE
Add `-` hotkey for toggling the signedness of IntegerDisplayType constants

### ISSUE THREAD

**Issue by bb33bb** [Vector35/binaryninja-api#6063]
**What is the feature you'd like to have?**
Please fix the shortcut of changing the type of number

**Is your feature request related to a problem?**
NO
**Are any alternative solutions acceptable?**
YES
by manual right click and select the corresponding button

**Additional Information:**
![image](https://github.com/user-attachments/assets/b61f3d79-30df-49c8-9bc8-8f1da83a6dca)

**Comment by bb33bb** [comment]
using The ➖(minus sign - ) button
is not effect.

**Comment by xusheng6** [comment]
> using The ➖(minus sign - ) button is not effect.

That is meant to toggle the signedness of a data variable, so it is not designed to work that way. However, your feature request is a valid one and worth considering

**Comment by bb33bb** [comment]
the latest version fixed this problem.

**Comment by xusheng6** [comment]
> the latest version fixed this problem.

I do not think we actually changed anything related recently. Is it because you tested with a data variable? If it is a data variable, pressing "-" on it will indeed toggle its signedness, and it will achieve the same result that you have expected. However, if the constant is from the code and not a data variable, I believe pressing "-" should not have worked. If that is the case, we should case this open until we actually support it

**Comment by bb33bb** [comment]
> > the latest version fixed this problem.
> 
> I do not think we actually changed anything related recently. Is it because you tested with a data variable? If it is a data variable, pressing "-" on it will indeed toggle its signedness, and it will achieve the same result that you have expected. However, if the constant is from the code and not a data variable, I believe pressing "-" should not have worked. If that is the case, we should case this open until we actually support it

老哥说的有道理，我确实测试快捷键有生效的了，但是测试另一部分，又出现不行的。我上传一个文档。
我重新打开这个，然后上传文件过来。

**Comment by xusheng6** [comment]
> > > the latest version fixed this problem.
> > 
> > 
> > I do not think we actually changed anything related recently. Is it because you tested with a data variable? If it is a data variable, pressing "-" on it will indeed toggle its signedness, and it will achieve the same result that you have expected. However, if the constant is from the code and not a data variable, I believe pressing "-" should not have worked. If that is the case, we should case this open until we actually support it
> 
> 老哥说的有道理，我确实测试快捷键有生效的了，但是测试另一部分，又出现不行的。我上传一个文档。 我重新打开这个，然后上传文件过来。

Thx for confirming! But we do not need a file for this, since it is quite easy to reproduce

**Comment by bb33bb** [comment]
> > > > the latest version fixed this problem.
> > > 
> > > 
> > > I do not think we actually changed anything related recently. Is it because you tested with a data variable? If it is a data variable, pressing "-" on it will indeed toggle its signedness, and it will achieve the same result that you have expected. However, if the constant is from the code and not a data variable, I believe pressing "-" should not have worked. If that is the case, we should case this open until we actually support it
> > 
> > 
> > 老哥说的有道理，我确实测试快捷键有生效的了，但是测试另一部分，又出现不行的。我上传一个文档。 我重新打开这个，然后上传文件过来。
> 
> Thx for confirming! But we do not need a file for this, since it is quite easy to reproduce

OK

**Comment by plafosse** [comment]
This should also work for toggling the sign of variables and parameters too.

**Comment by pjcreath** [comment]
As I see it, the current limitation is that the only commands exposed (via command palette, Display As context menu, etc.) are ones that select a specific type.  Exposing commands to toggle particular variations would be great, not just for signedness but also hex/decimal and bitwise-complement:

* Toggle between signed and unsigned
* Toggle between hex and decimal
* Toggle between bitwise complement and normal

All three of these attributes are theoretically independent, which is how IDA handles them (with the hotkeys "-", "h", and "~" respectively).  You could probably implement the first two without changes to your data model by simply selecting the appropriate IntegerDisplayType depending on its current type.

This last one is potentially a separate feature request, since there's no current way to display the bitwise complement of a constant.  But it's hugely useful to be able to display a 0xffffffe0 mask as "~0x1f".  Poking at it a little bit, it seems that IDA considers bitwise complement to be unsigned, so you'd only need to add bitwise complement hex and bitwise complement decimal.
